py_pdf_parser/components.py

```diff
@@ -338,6 +338,9 @@ class PDFDocument:
     sectioning: "Sectioning"
     # _element_list will contain all elements, sorted from top to bottom, left to right.
     _element_list: List[PDFElement]
+    # _elements_indexes_by_font will be a caching of fonts to elements indexes but it
+    # will be built as needed (while filtering by fonts), not on document load.
+    _elements_indexes_by_font: Dict[str, Set[int]]
```

Comment:
This reads a bit strange to me, I think you've basically done a double plural on the same thing.

```suggestion
    # _element_indexes_by_font will be a caching of fonts to elements indexes but it
    # will be built as needed (while filtering by fonts), not on document load.
    _element_indexes_by_font: Dict[str, Set[int]]
```